shotover-proxy/src/config/chain.rs

```diff
@@ -0,0 +1,89 @@
+use crate::transforms::chain::TransformChainBuilder;
+use crate::transforms::{TransformBuilder, TransformConfig};
+use anyhow::Result;
+use serde::de::{DeserializeSeed, Deserializer, MapAccess, SeqAccess, Visitor};
+use serde::Deserialize;
+use std::fmt::{self, Debug};
+use std::iter;
+
+#[derive(Deserialize, Debug)]
+pub struct TransformChainConfig(
+    #[serde(rename = "TransformChain", deserialize_with = "vec_transform_config")]
+    pub  Vec<Box<dyn TransformConfig>>,
+);
+
+impl TransformChainConfig {
+    pub async fn get_builder(&self, name: String) -> Result<TransformChainBuilder> {
+        let mut transforms: Vec<Box<dyn TransformBuilder>> = Vec::new();
+        for tc in &self.0 {
+            transforms.push(tc.get_builder(name.clone()).await?)
+        }
+        Ok(TransformChainBuilder::new(transforms, name))
+    }
+}
+
+fn vec_transform_config<'de, D>(deserializer: D) -> Result<Vec<Box<dyn TransformConfig>>, D::Error>
```

Comment:
I think it would be worth a short comment explaining why we need to do our own Deserialiser stuff here.